posthog/temporal/batch_exports/backfill_batch_export.py

```diff
@@ -354,8 +369,18 @@ async def run(self, inputs: BackfillBatchExportInputs) -> None:
             retry_policy=temporalio.common.RetryPolicy(maximum_attempts=0),
         )
 
-        backfill_duration = dt.datetime.fromisoformat(inputs.end_at) - dt.datetime.fromisoformat(inputs.start_at)
-        number_of_expected_runs = backfill_duration / dt.timedelta(seconds=frequency_seconds)
+        # Temporal requires that we set a timeout.
+        if inputs.end_at is None:
+            # Set timeout to a month for now, as unending backfills are an internal feature we are
+            # testing for HTTP-based migrations. We'll need to pick a more realistic timeout
+            # if we release this to customers.
+            start_to_close_timeout = dt.timedelta(days=31)
+        else:
+            # Allocate 5 minutes per expected number of runs to backfill as a timeout.
+            # The 5 minutes are just an assumption and we may tweak this in the future
+            backfill_duration = dt.datetime.fromisoformat(inputs.end_at) - dt.datetime.fromisoformat(inputs.start_at)
+            number_of_expected_runs = backfill_duration / dt.timedelta(seconds=frequency_seconds)
+            start_to_close_timeout = dt.timedelta(minutes=5 * number_of_expected_runs)
```

Comment:
I'm thinking whether it would make sense to refactor this and have the workflow issue one activity per batch period instead. It would obviously be more costly (perhaps too costly), but I think it would make sense from a timeout/retrying perspective. Ah well, a month is okay for starters.